From a71f0ce3e2dda3fa0fab91a9d480cea708fe4d95 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 2 Aug 2026 13:38:23 +0200
Subject: [PATCH] ARM: dts: allwinner: sun8i-a83t-tbs-a711: Add speaker
 amplifier supply

The speaker amplifier is powered from VCC-5V, the rail a DC-DC converter
makes from PS and that also feeds USB, already modeled here as the
vmain regulator. Without the supply the driver uses a dummy regulator:

  simple-amplifier audio-amplifier: supply VCC not found, using dummy
  regulator

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
index 1ae04c1219d3..0d029f57fc02 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
@@ -179,6 +179,7 @@ bt_sco_codec: bt-sco-codec {
 	speaker_amp: audio-amplifier {
 		compatible = "simple-audio-amplifier";
 		enable-gpios = <&pio 6 13 GPIO_ACTIVE_HIGH>; /* PG13 */
+		VCC-supply = <&reg_vmain>;
 		sound-name-prefix = "Speaker Amp";
 	};
 
-- 
2.43.0

